app/models/task.py

```diff
@@ -2,4 +2,29 @@
 
 
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String, nullable=False)
+    description = db.Column(db.String)
+    completed_at = db.Column(db.DateTime, nullable=True)
+    goal_id = db.Column(db.Integer, db.ForeignKey("goal.goal_id"))
+    goal = db.relationship("Goal", back_populates="tasks", lazy=True)
+
+    def is_complete(self):
+        return self.completed_at is not None
+
+    def to_dict(self):
+        
+        if self.goal_id is None:
+            return {"id": self.id,
+                    "title": self.title,
+                    "description": self.description,
+                    "is_complete": self.is_complete()
+            }
+        else:
+            return {"id": self.id,
+                    "goal_id": self.goal_id,
+                    "title": self.title,
+                    "description": self.description,
+                    "is_complete": self.is_complete()
+            }
```

Comment:
Stylistically, I recommend rewriting this section as:
```suggestion
        output = {"id": self.id,
                    "title": self.title,
                    "description": self.description,
                    "is_complete": self.is_complete()
            }
        if self.goal_id is not None:
            output["goal_id" = self.goal_id]
        return output
```
The structure of the dictionary is the same in both if branches, the only difference is that goal_id is included if it's not None. As written, it requires a much closer reading of the code to determine this. In addition, pulling the sections that are the same out of the if block makes it possible to DRY the code.